### ISSUE TITLE
Provide a variable similar to `$PSStyle` for Emoji

### ISSUE THREAD

**Issue by jpawlowski** [PowerShell/PowerShell#16720]
### Summary of the new feature / enhancement

It would be great to have built-in support for easy access to unicode items using short codes.

### Proposed technical implementation details (optional)

The new `$PSStyle` variable is great and there could be something similar like `$PSEmoji` for common Emojis and maybe even a selected source of other Unicode letters in `$PSUnicode`.

Short codes for Emojis shall be based on existing GitHub short codes and could be based on the [GitHub Emoji API](https://api.github.com/emojis) together with [Unicode Full Emoji List](https://unicode.org/emoji/charts/full-emoji-list.html), similar to what is implemented by @ikatyang in his [Emoji Cheat Sheet](https://github.com/ikatyang/emoji-cheat-sheet/).

**Comment by rkeithhill** [comment]
Have you tried this module from the PowerShell Gallery? https://www.powershellgallery.com/packages/Emojis/0.1

![image](https://user-images.githubusercontent.com/5177512/148702440-f588937d-c2d4-415e-b491-188ce7f033ed.png)

**Comment by jpawlowski** [comment]
I know this module but was looking for a built-in/native functionality to be potentially more lightweight. Also, I would like to avoid external dependencies as much as possible. I believe `$PSStyle` was born for similar reasons (and others I can't imagine).

**Comment by MartinGC94** [comment]
What would be the purpose of this? Is it to make it easier to write emojis into a script/console? Or is it to have a different, more descriptive way to add them to a script?  
I don't use emojis in scripts, but if I did I would imagine it's easier to quickly read and write `Write-Host "It worked! ✔✔✔" ` than `Write-Host "It worked $(PSEmoji.GreenCheckMark)$(PSEmoji.GreenCheckMark)$(PSEmoji.GreenCheckMark)"`

Windows includes a handy shortcut: `Win+.` to quickly add emojis in any text box and I would imagine OS X + Most desktop Linux distros have a similar feature.

**Comment by jpawlowski** [comment]
I don't use Windows, the Mac is all I need and provides Emojis for a long time using keyboard shortcuts 🙃 </OT>

To me, the purpose is everything you just wrote above. I think it is still not best practice to add UTF8 characters directly into text files/scripts, even though they might be UTF8 encoded. Am I wrong?

**Comment by jborean93** [comment]
While it doesn't have any autocompleteion support you can simply use the `u` escape sequence.

```powershell
"`u{1F3B5}" # 🎵 https://www.compart.com/en/unicode/U+1F3B5
"`u{1F4A9}" # 💩 https://www.compart.com/en/unicode/U+1F4A9
```

Essentially it's `<backtick>u{unicode codpoint}`

> `u{unicode codepoint}

**Comment by jpawlowski** [comment]
Thank you, I know I can use direct UTF8 characters. The point is I need to look them up on a case by case basis and the pure number does not tell anyone later on what’s behind. Also, this is redundant code that usually could be avoided.

I can maintain a large hash table for each and every module, yes. Again, this is redundant code and better centrally maintained and kept up-to-date.

**Comment by MartinGC94** [comment]
I don't know what the best practices are for special characters in scripts because I usually avoid them. If I had to include special characters I would only do something special if they are hard to distinguish from "real" characters (em dash VS regular dash).  
Since Emojis can easily be read on their own I would just add them directly inside whatever string I'm using.  
One could also argue that using emojis in a script in the first place goes against best practices because regardless of the script file encoding you can have issues rendering it in the console or whatever it is you are outputting to:
![image](https://user-images.githubusercontent.com/42123497/148707294-5ca10b57-a5c5-4431-b8de-7527d37d2ea5.png)

The point I was making before was that I think something like `$PSEmoji.GreenCheckMark` is a worse solution than just writing the emoji directly into the script and since Windows and apparently also OS X includes an easy way to type them in then I see little point in this feature.

**Comment by jpawlowski** [comment]
All what's been said is true for ANSI characters as well, which is why PSStyle was introduced.

Also, there is now the new Windows Terminal becoming the de-facto standard to wrap the PowerShell host. Displaying issues are no more existing (just catching up with macOS and Linux, okay).

---
TL;DR

I really regret having opened this Feature Request.
This is not a forum (GitHub has the discussion section for this). Pointing me to obvious things I already know do not help with the suggestion I made. Also, I feel like it is only worth commenting if you are part of the maintainers, which does not seem to be the case for some commenters.

I forgot, this is the internet and everyone has to comment on everything.

Should there be further PS developer feedback on this, let me know; I won't answer any other comments anymore. #trollsoff

**Comment by iSazonov** [comment]
To implement the request internally we need to see:
1. Either important business scenario(s)
2. or many votes from community
3. or if something such already implemented in other (1) shells or (2) script languages.

**Comment by jpawlowski** [comment]
To me, the scenario is fairly similar to the existence of `$PSStyle`.
You can close the FR if you disagree and we just leave this documented here.
